### PR TITLE
CI improvements

### PR DESCRIPTION
**Pull request by nicoddemus**
Improvements based on #905:

* Fix tag creation in deploy workflow
* Use same package in all test runs

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -43,5 +43,7 @@ jobs:
 
     - name: Push tag
       run: |
+        git config user.name "pytest bot"
```
Comment (bluetech) [inline .github/workflows/deploy.yml:46]: Oops forgot about this :+1:

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -68,14 +70,24 @@ jobs:
...
     - name: Install tox
       run: |
         python -m pip install --upgrade pip
         pip install tox
+
     - name: Test
+      shell: bash
       run: |
-        tox -e ${{ matrix.tox_env }}
+        tox run -e ${{ matrix.tox_env }} --installpkg `find dist/*.tar.gz`
```
Comment (bluetech) [inline .github/workflows/test.yml:93]: Testing the actual package (alike) we intent to ship is a good pattern, and from the tox output seems to work as intended.

**Review by bluetech — APPROVED** [review]

File: .github/workflows/test.yml (hunk 2 above)
Comment (RonnyPfannschmidt) [reply .github/workflows/test.yml:93]: at first glance we only test the sdist and not the wheel

**Review by RonnyPfannschmidt — APPROVED** [review]

File: .github/workflows/test.yml (hunk 2 above)
Comment (nicoddemus) [reply .github/workflows/test.yml:93]: Indeed, but AFAIK this is how it always has been: tox generates both wheel and sdist, but only installs sdist:

```
λ tox -e py310
...
py310: install_package_deps> python -I -m pip install execnet>=1.1 filelock pytest>=6.2.0
py310: install_package> python -I -m pip install --force-reinstall --no-deps E:\projects\pytest-xdist\.tox\.tmp\package\4\pytest-xdist-3.3.1.dev2+g6d39025.tar.gz
py310: commands[0]> pytest
...
```

Testing both seems a bit overkill, if that's what you are suggesting.

**Comment by nicoddemus** [comment]
Btw, I could swear I have left a comment regarding this, but here it goes:

I think the next step is to have a another job which depends on `test`, and does the deployment to PyPI under a few conditions:

1. From a branch in the format `release-*`, under pytest-dev.
2. Also using the `deploy` environment, meaning that this job will only start after manual approval.

What do you folks think?

**Pull request merged**